junit-platform-engine/src/main/java/org/junit/platform/engine/support/hierarchical/EngineExecutionContext.java

```diff
@@ -22,5 +22,15 @@
  * @see HierarchicalTestEngine
  */
 @API(status = MAINTAINED, since = "1.0")
-public interface EngineExecutionContext {
+public interface EngineExecutionContext extends AutoCloseable {
+	/**
+	 * Closes this context, close any underlying resources.
+	 *
+	 * <p>This default implementation does nothing.
+	 *
+	 * @since 1.1
+	 */
+	@Override
+	default void close() {
```

Comment:
I think we should allow throwing exceptions from implementations of `close()`.

Reply:
Ok.

Reply:
Here. I removed `extends AutoCloseable` and `@Override` ... to avoid a warning emitted by `javac`.